README.md

```diff
@@ -0,0 +1,319 @@
+
+## Blocks
+
+The store builder allows you to extend existing blocks, like the ones provided by the Dream Store, allowing them to replace the blocks they extend via the Storefront editor. They also allow you to build completely new blocks that provide functionality specific to your business needs and, then, create templates to insert those new blocks in your client's store while still benefiting from the continuous progress made by the Dream Store and it's extensions on the other parts of your store.
+
+### Providing new templates
+
+Let's build an app that provides a new home page template. Our template will be available on the Storefront editor for the admin to select among the various templates available. The Dream Store provides a home page template and the various blocks used in it, among several other templates and blocks, let's use the available blocks to build a home page different from the Dream Store's home page.
+
+First, we will need to [build a new IO app](http://help.vtex.com/en/tracks/vtex-io-getting-started). Let's call our app `partner.theme`. The starting point to creating a new template is the `store` builder. First, let's use that by declaring it on our app's `manifest.json`.
+
+```jsonc
+{
+    "vendor": "partner",
+    "name": "theme",
+    // other configuration
+    "builders": {
+        // other builders
+        "store": "0.x"
+    }
+}
+```
+
+Configuring the store builder requires a `store` folder on the root of our app. If we want to create templates, we will need a `blocks.json` file inside that `store` folder. This means that, for now, our app's folder structure contains at least:
+
+```txt
+.
+├── manifest.json
+└── store
+    └── blocks.json
+```
+
+The Dream Store has already defined a home page template, which is the one used by default. Let's take a look at what that portion of the `blocks.json` from the `vtex.dreamstore` app looks like.
+
+```jsonc
+{
+    "store/home": {
+        "name": "Dream Store home page",
+        "blocks": ["header/full", "banner", "shelf", "footer"],
+    },
+    // some other blocks
+}
+```
+
+This file is declaring a `store/home` block which has a descriptive `name` which will be displayed in the Storefront editor. It also declares which blocks are contained within it. This means that the same `blocks.json` may declare those blocks or that the app depends on apps that declare those blocks inside their `blocks.json` files. In this specific case, the `vtex.dreamstore` app depends on apps that declare each of those, `vtex.header` declaring `header`, `vtex.banner` declaring `banner`, `vtex.shelf` declaring `shelf` and `vtex.footer` declaring `footer`.
+
+Why did Dream Store use `store/home` instead of just calling the block `home`? This means that `store/home` is a block that **extends** a `store` block. The `vtex.dreamstore` app depends on `vtex.store` which declares the `store` block, let's take a look at that declaration.
```

Comment:
1. It won't be able to fill the home page because that route is declared in `vtex.store` and refers to `store/home`, not to `home`.
2. The `store/home` block also defined in `vtex.store` gives a context provider.
3. We also won't allow people to declare routes that don't extend `store` for now.